app/src/androidTest/java/com/github/multimatum_team/multimatum/MainActivityTest.kt

```diff
@@ -181,6 +186,46 @@ class MainActivityTest {
         }
     }
 
+    /*
+    The two following helper functions have been found online at:
+    https://alexzh.com/ui-testing-of-android-runtime-permissions/
+     */
+    private fun grantPermission() {
+        val instrumentation = InstrumentationRegistry.getInstrumentation()
+        if (Build.VERSION.SDK_INT >= 23) {
+            val allowPermission = UiDevice.getInstance(instrumentation).findObject(
+                UiSelector().text(
+                    when {
+                        Build.VERSION.SDK_INT == 23 -> "Allow"
+                        Build.VERSION.SDK_INT <= 28 -> "ALLOW"
+                        Build.VERSION.SDK_INT == 29 -> "Allow only while using the app"
+                        else -> "While using the app"
+                    }
+                )
+            )
+            if (allowPermission.exists()) {
+                allowPermission.click()
+            }
+        }
+    }
+
+    private fun denyPermission() {
+        val instrumentation = InstrumentationRegistry.getInstrumentation()
+        if (Build.VERSION.SDK_INT >= 23) {
```

Comment:
Same as previous comment, `minSdk` is 26

Reply:
Right